core/src/main/java/com/linecorp/armeria/server/ServiceConfig.java

```diff
@@ -392,6 +400,13 @@ public Path multipartUploadsLocation() {
         return multipartUploadsLocation;
     }
 
+    /**
+     * Returns the {@link Function} that generates a {@link RequestId}.
+     */
+    public Function<? super RoutingContext, ? extends RequestId> requestIdGenerator() {
```

Comment:
The return type is different from others that are invariant types.
Variance for a function is useful for an input parameter for flexibility but nothing to gain from return types.
```suggestion
    public Function<RoutingContext, RequestId> requestIdGenerator() {
```